extensions/amp-apester-media/0.1/monetization/companion/video.js

```diff
@@ -0,0 +1,151 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Services} from '../../../../../src/services';
+
+/**
+ * @param {!JsonObject} media
+ * @param {AmpElement} apesterElement
+ * @param {!JsonObject} consentObj
+ * @return {Promise}
+ */
+export function handleCompanionVideo(media, apesterElement, consentObj) {
+  const monetizationSettings = media['campaignData'] || {};
+  const companionCampaignOptions =
+    monetizationSettings['companionCampaignOptions'] || {};
+  const companionRawSettings = monetizationSettings['companionOptions'] || {};
+  if (
+    companionRawSettings.video &&
+    companionRawSettings.video.enabled &&
+    companionRawSettings.video.provider === 'sr'
+  ) {
+    const position = getCompanionPosition(companionRawSettings.video);
+    if (position && position !== 'floating') {
+      const macros = getSrMacros(
+        media,
+        companionCampaignOptions['companionCampaignId'],
+        apesterElement,
+        consentObj
+      );
+
+      constructCompanionSrElement(
+        companionRawSettings['video']['videoTag'],
+        position,
+        /** @type {!JsonObject} */ (macros),
+        apesterElement
+      );
+    }
+  }
+  return Promise.resolve();
+}
+/**
+ * @param {!JsonObject} video
+ * @return {string}
+ */
+function getCompanionPosition(video) {
+  if (video['companion']['enabled']) {
+    return 'above';
+  } else if (video['companion_below']['enabled']) {
+    return 'below';
+  } else if (video['floating']['enabled']) {
+    return 'floating';
+  }
+  return '';
+}
+
+/**
+ * @param {string} videoTag
+ * @param {string} position
+ * @param {!JsonObject} macros
+ * @param {AmpElement} apesterElement
+ */
+function constructCompanionSrElement(
+  videoTag,
+  position,
+  macros,
+  apesterElement
+) {
+  const size = getCompanionVideoAdSize(apesterElement);
+  const ampAd = apesterElement.ownerDocument.createElement('amp-ad');
+  ampAd.setAttribute('type', 'blade');
+  ampAd.setAttribute('data-blade_player_type', 'bladex');
+  ampAd.setAttribute('servingDomain', 'ssr.streamrail.net');
+  ampAd.setAttribute('width', size.width);
+  ampAd.setAttribute('height', size.height);
+  ampAd.setAttribute('data-blade_macros', JSON.stringify(macros));
+  ampAd.setAttribute('data-blade_player_id', videoTag);
+  ampAd.setAttribute('data-blade_api_key', '5857d2ee263dc90002000001');
+  ampAd.classList.add('amp-apester-companion');
+
+  const relativeElement =
+    position === 'below' ? apesterElement.nextSibling : apesterElement;
+
+  apesterElement.parentNode.insertBefore(ampAd, relativeElement);
+}
+
+/**
+ * @param {AmpElement} apesterElement
+ * @return {{height: number, width: number}}
+ */
+function getCompanionVideoAdSize(apesterElement) {
+  const adWidth = apesterElement./*REVIEW*/ clientWidth;
```

Comment:
This feels weird. Is this supposed to be a static size? What happens if the element or its container is resized smaller?

Does`layout="responsive"` with `width="10" height="6"` make sense here?

Reply:
@sparhami 

No,

it's static because the ad width should be equal to Apester unit width.

Reply:
Can the Apester unit change width the viewport? Based on the component, it can, since it supports all sized defined layouts, including responsive. If this element was placed within apesterElement, the width would always match if you use `layout="responsive"`.